### DISCUSSION TITLE
Equivalent functionality to `tfjs` `memory`

### DISCUSSION THREAD

**Discussion by sklum** [microsoft/onnxruntime#20236] [Web Q&A]
`Tensorflow.js` has the [`memory`](https://js.tensorflow.org/api/latest/#memory) functionality which makes it very easy to test if you're leaking memory via tensor allocations in a given function like this:

```
  const numTensorsBefore = memory().numTensors;
  await f();
  expect(memory().numTensors - numTensorsBefore).toEqual(0);
```

Does `onnxruntime-web` have any similar functionality? It didn't look like the profiling used [in the unit tests](https://github.com/microsoft/onnxruntime/blob/main/js/web/test/test-runner.ts#L201) would give this level of information, and I was not able to find other examples of this sort of thing or anything in the documentation.

**Comment by carzh** [comment]
No, we don't have this kind of functionality in onnxruntime-web, as far as I know

**Reply by sklum** [reply]
Thanks @carzh.  More generally, do you have any suggestions for checking for memory leaks?  It seems like https://github.com/microsoft/onnxruntime/blob/main/js/common/lib/tensor.ts#L63 needs to be called from https://github.com/microsoft/onnxruntime/blob/main/js/web/lib/wasm/wasm-core-impl.ts#L635, but that wasn't clear to me from the docs, so I just want to make sure I'm not leaking memory elsewhere.
